File: 13.roman-to-integer.py
#
# @lc app=leetcode id=13 lang=python3
#
# [13] Roman to Integer
#

# @lc code=start
class Solution:
    def romanToInt(self, s: str) -> int:
        values = {
            'I':1,
            'V':5,
            'X': 10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000
}
        i =0
        total = 0 
        while i < len(s):
            if  i < (len(s)-1) and values[s[i]]< values [s[i+1]]:
                total += values[s[i+1]] - values[s[i]]
                i +=2
            else:
                total +=values[s[i]]
                i+=1
        return total
        
# @lc code=end

File: 13.roman-to-integer.java
/*
 * @lc app=leetcode id=13 lang=java
 *
 * [13] Roman to Integer
 */

// @lc code=start
class Solution {
    public int romanToInt(String s) {

        int ans = 0 , num =0;
        for (int i = 0; i < s.length(); i++) {
        char c = s.charAt(s.length() - 1 - i);
        switch (c) {
            case 'I':
                num = 1;
                break;
            case 'V':
                num = 5;
                break;
            case 'X':
                num = 10;
                break;
            case 'L':
                num = 50;
                break;
            case 'C':
                num = 100;
                break;
            case 'D':
                num = 500;
                break;
            case 'M':
                num = 1000;
                break;
        }
        if (4*num < ans) {
            ans -=num;
        }
        else{
            ans +=num;
        }
    }
        
    return ans;
        
        
    
}
}
// @lc code=end

